### ISSUE TITLE
CKANSink refactoring caused recvTimeTs go out of range

### ISSUE THREAD

**Issue by reitsma** [telefonicaid/fiware-cygnus#1947]
In commit d50030e8d776de26a157663fd170ad4a610a4b06 the CkanSink has been refactoring by moving the aggregation to a generic class. However, an important thing is overlooked: In CKAN the recvTimeTs  in in seconds, otherwise it does not fit in the PostgreSQL integer field (4 bytes). 

This is before:
NGSIConstants.RECV_TIME_TS + "\": \"" + recvTimeTs / 1000

After:
aggregation.get(NGSIConstants.RECV_TIME_TS).add(new JsonPrimitive(Long.toString(recvTimeTs)))

**Comment by fgalan** [comment]
Thanks for the report! We'll have a look

CC: @IvanHdzC

**Comment by diego-gv** [comment]
Hi all,

I am trying to use CKANSink, but this problem seems to persist. 

I am using version 2.11.0 via docker, in any of the following versions has this bug been fixed?

Best regards.

**Comment by fgalan** [comment]
> 
> 
> Hi all,
> 
> I am trying to use CKANSink, but this problem seems to persist.
> 
> I am using version 2.11.0 via docker, in any of the following versions has this bug been fixed?
> 
> Best regards.

Looking to the changelog delta from 2.11.0 to 2.14.0 (current release) I see some fixes related with CKAN. Not sure if they include the issue described here, but, anyway, I think is a good idea to test with the last version and, if the issues persists, then fix is there.

@diego-gv could you teset with the newest version and tell us, please? Thanks!

**Comment by diego-gv** [comment]
> Looking to the changelog delta from 2.11.0 to 2.14.0 (current release) I see some fixes related with CKAN. Not sure if they include the issue described here, but, anyway, I think is a good idea to test with the last version and, if the issues persists, then fix is there.
> 
> @diego-gv could you teset with the newest version and tell us, please? Thanks!

Hello @fgalan

It is certainly a good idea and it is the next action we took immediately after posting the comment (build the image of the following versions, as they did not exist) but unfortunately the problem persists.

Regards,

**Comment by fgalan** [comment]
> It is certainly a good idea and it is the next action we took immediately after posting the comment (build the image of the following versions, as they did not exist) but unfortunately the problem persists.

Do you mean that you have tested with Cygnus 2.14.0 and the problem persist? (Just to have it fully clear :)

**Comment by diego-gv** [comment]
Yes, I tried with newer versions (version 2.14.0 included) but the problem persists.

**Comment by fgalan** [comment]
@diego-gv thanks for your feedback!

By the moment this issue is not in our priorities, but it would be great if somebody in the community would like to have a look and solve it. We could provide support to that effort reviewing the pull request with the fix and giving feedback on it :)
